lib/live_ex.ex

```diff
@@ -26,32 +29,43 @@ defmodule LiveX do
       @doc """
       Dispatch an Action with a `type` and an optional payload.
       """
-      def dispatch(type, payload \\ %{}, socket) do
-        event = %{
-          type: String.to_atom(type),
-          payload: payload
-        }
+      def dispatch(type, payload \\ nil, socket)
+
+      @spec dispatch(atom, any, socket) :: map
+      def dispatch(type, payload, socket) when is_atom(type) do
+        action = %{type: type, payload: payload}
 
-        send(socket.assigns.pid, event)
+        send(socket.assigns.pid, action)
+      end
+
+      @spec dispatch(String.t(), any, socket) :: map
+      def dispatch(type, payload, socket) do
+        type
+        |> String.to_atom()
+        |> dispatch(payload, socket)
       end
 
       @doc """
       Commit a change to the store.
 
 
       """
+      @spec commit(atom, any, socket) :: {:noreply, socket}
       def commit(type, payload, socket) do
-        log(type, payload, socket, fn ->
-          apply(__MODULE__, type, [payload, socket])
-        end)
+        socket =
+          log(type, payload, socket, fn ->
+            apply(__MODULE__, type, [payload, socket])
+          end)
+
+        {:noreply, socket}
       end
 
       defp log(type, payload, socket, fun) do
         state_before = socket.assigns
         socket = fun.()
         state_after = socket.assigns
 
-        Logger.debug("LiveX
+        Logger.debug("LiveEx
```

Comment:
As log messages are not necessary displayed in the same order as the log statements are invoked the usefulness in terms of reenacting state changes this way is limited. In case you are trying to debug state changes erlang already has you covered with `:sys.trace/2`

Reply:
Good point. I looked into `:sys.trace/2` and it seems like I need to start the `Store` in its own Process in order to trace its function calls. Moving the `Store` to its own Process is on the roadmap for after the Unconference and I will address this issue then.

Reply:
Addressed in #3